doit/dependency.py

```diff
@@ -426,6 +426,26 @@ def get_state(self, dep, current_state):
             'timestamp': TimestampChecker}
 
 
+class DependencyStatus(object):
+    """Result object for Dependency.get_status.
+
+    @ivar status: (str) one of "run", "up-to-date" or "error"
+    """
+
+    def __init__(self):
+        self.status = None
+
+    def _decide(self, status):
```

Comment:
I now made the `decide()` method local to `get_status` (instead of having it in `DependencyStatus`). Now the only drawback is that `get_status` sets `DependencyStatus._error_message` (which is read by `DependencyStatus.get_error_message()`.